src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -88,6 +94,30 @@ private ConfigurationSetting<IList<ManifestInfo>> GetDefaultManifestInfoForValid
             };
         }
 
+        private ConfigurationSetting<string> GetTempBuildDropPath(IConfiguration configuration)
+        {
+            if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value != null && configuration.DockerImagesToScan?.Value != null)
+            {
+                return new ConfigurationSetting<string>
+                {
+                    Source = SettingSource.Default,
+                    Value = IConfiguration.RandomTempPath,
+                };
+            }
+            else if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value == null && configuration.BuildComponentPath?.Value == null && configuration.DockerImagesToScan?.Value != null)
+            {
+                throw new ValidationArgException($"Please provide a (-m) if you intend to create an SBOM with only the contents of the Docker image or a (-bc) if you intend to include other components in your SBOM.");
+            }
+            else if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value == null && configuration.DockerImagesToScan?.Value != null)
+            {
+                throw new ValidationArgException($"Please provide a value for the ManifestDirPath (-m) parameter to generate the SBOM for the specified Docker image.");
+            }
+            else
+            {
+                throw new ValidationArgException($"Please provide a value for the BuildDropPath (-b) parameter to generate the SBOM.");
```

Comment:
I would create a separate method to validate the configuration and call it before calling GetTempBuildDropPath

Reply:
Fixed.